packages/core/useCloned/index.ts

```diff
@@ -0,0 +1,43 @@
+import type { MaybeRef } from '@vueuse/shared'
+import type { WatchOptions } from 'vue-demi'
+import { isRef, ref, unref, watch } from 'vue-demi'
+
+export interface UseClonedOptions<T extends Record<any, any> = Record<any, any>> {
```

Comment:
```suggestion
export interface UseClonedOptions<T> extends WatchOptions {
```

We'd better directly support watch options instead of nesting, to align with other functions